tests/tests.json

```diff
@@ -0,0 +1,713 @@
+{
+  "base-uri": "coaps://foo:4711/pa/th?query#frag",
+  "base-cri": "85218263666f6f19126782627061627468816571756572796466726167",
+  "test-vectors": [
+    {
+      "uri": "",
+      "cri": "8100",
+      "uri-from-cri": "",
+      "resolved-cri": "85218263666f6f19126782627061627468816571756572796466726167",
+      "resolved-uri": "coaps://foo:4711/pa/th?query#frag"
+    },
+    {
+      "uri": "//a",
+      "cri": "82f66161",
+      "uri-from-cri": "//a",
+      "resolved-cri": "82216161",
+      "resolved-uri": "coaps://a"
+    },
+    {
+      "uri": "//192.168.0.97",
+      "cri": "82f644c0a80061",
```

Comment:
is `[null, h'C0A80061']`
should be `[null, [h'C0A80061']]`

Reply:
The missing array around authority seems to be the only problem my tests find.
(Of course, we could add these cases to the syntax, saving one byte.)

One limitation is that the test vectors don't have a domain name with a dot.  So I can't test that aspect.

Do you want me to merge this or do you want to fix the array problem first?

Reply:
```suggestion
      "cri": "82f68144c0a80061",
```

Reply:
Well, this wasn't just this one test...
I find 37 more tests hat need this change.